src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -125,6 +125,77 @@ private static void SendToRecvFromAsync_Datagram_UDP(IPAddress leftAddress, IPAd
             }
         }
 
+        private static void SendToRecvFromAsync_UdpClient_Datagram_UDP(IPAddress leftAddress, IPAddress rightAddress)
```

Comment:
Nit: for clarity we usually split the tests first by class, not async technology. I would place these tests in separate UdpClient / TcpClient test files.
One other reason is if we ever want to test only the 4.0 contract where these types don't exist (e.g. some security fix).